hydra-main/src/main/java/com/addthis/hydra/query/aggregate/QueryTask.java

```diff
@@ -122,6 +85,57 @@ public void run() {
         }
     }
 
+    private int readFrame(List<QueryTaskSource> taskSources, int maxReads) throws Exception {
```

Comment:
"readFrame" is kind of a misleading name here. That might be appropriate for FramedDataChannelReader... but that just has `read` for interface purposes.